### PR TITLE
Spelling terms cleanup.

### PR DESCRIPTION
**Pull request by mk46**
Signed-off-by: Manish Kumar <manish.kumar1@india.nec.com>

Commit Message: Spelling terms cleanup.
Additional Description: Cleaning `tools/spelling/spelling_dictionary.txt` which are single-use keywords.
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
Includes a partial fix to #12807

### CODE REVIEW

**Review by ggreenway — COMMENTED** [review]
Some CI jobs are failing

/wait

**Comment by mk46** [comment]
I had checked the failed CI, clang-tidy error on WIP  `tools/clang_tools/api_booster/main.cc`  and another in error in  `source/extensions/quic_listeners/quiche/platform/quic_pcc_sender_impl.h`  due to wrong naming convention for method `CreatePccSenderImpl` .
Is it okay rename `CreatePccSenderImpl` to `createPccSenderImpl` ?

File: tools/spelling/spelling_dictionary.txt [hunk 1]
```diff
@@ -201,53 +188,34 @@ NOLINT
...
-PostCBs
 PREBIND
 PRNG
 PROT
-Postgre
 Postgres
-Prereq
 QDCOUNT
 QUIC
-QoS
```
Comment (ggreenway) [inline tools/spelling/spelling_dictionary.txt:241]: QoS is widely understood and should be left

File: source/common/signal/signal_action.h [hunk 2]
```diff
@@ -18,10 +18,10 @@ namespace Envoy {
  *
  * These signals are handled:
  *   SIGABRT
```
Comment (ggreenway) [inline source/common/signal/signal_action.h:20]: Why isn't this one quoted? It should be consistent with the rest of them, even if the spell checker doesn't require it.

File: tools/spelling/spelling_dictionary.txt [hunk 3]
```diff
@@ -261,52 +229,33 @@ RSA
 RST
 RTDS
 RTTI
-RUNDIR
-RW
```
Comment (ggreenway) [inline tools/spelling/spelling_dictionary.txt:265]: commonly used and understood. Please revert.

File: source/extensions/quic_listeners/quiche/platform/quic_logging_impl.cc [hunk 4]
```diff
@@ -32,7 +32,7 @@ QuicLogEmitter::~QuicLogEmitter() {
   }
   std::string content = stream_.str();
   if (!content.empty() && content.back() == '\n') {
-    // strip the last trailing '\n' because spd log will add a trailing '\n' to
+    // strip the last trailing '\n' because `spd` log will add a trailing '\n' to
```
Comment (ggreenway) [inline source/extensions/quic_listeners/quiche/platform/quic_logging_impl.cc:35]: change to `spdlog`; it shouldn't be two words

File: source/extensions/transport_sockets/tls/utility.cc [hunk 5]
```diff
@@ -37,9 +37,10 @@ std::string getRFC2253NameFromCertificate(X509& cert, CertName desired_name) {
 
   // flags=XN_FLAG_RFC2253 is the documented parameter for single-line output in RFC 2253 format.
   // Example from the RFC:
-  //   * Single value per Relative Distinguished Name (RDN): CN=Steve Kille,O=Isode Limited,C=GB
-  //   * Multivalue output in first RDN: OU=Sales+CN=J. Smith,O=Widget Inc.,C=US
-  //   * Quoted comma in Organization: CN=L. Eagle,O=Sue\, Grabbit and Runn,C=GB
+  //   * Single value per Relative Distinguished Name (`RDN`): CN=Steve `Kille`,O=`Isode`
```
Comment (ggreenway) [inline source/extensions/transport_sockets/tls/utility.cc:40]: The entire portion after the `:` should be in quotes, not individual portions of it

File: source/server/connection_handler_impl.cc [hunk 6]
```diff
@@ -238,7 +238,7 @@ ConnectionHandlerImpl::ActiveTcpListener::~ActiveTcpListener() {
   // being posted. This assert is extremely useful for debugging the common path so we will leave it
   // for now. If it becomes a problem (developers hitting this assert when using debug builds) we
   // can revisit. This case, if it happens, should be benign on production builds. This case is
-  // covered in ConnectionHandlerTest::RemoveListenerDuringRebalance.
+  // covered in `ConnectionHandlerTest::RemoveListenerDuringRebalance`.
```
Comment (ggreenway) [inline source/server/connection_handler_impl.cc:241]: I think It's fine to leave rebalance in the dictionary. It's close enough to a real word.

File: test/common/buffer/buffer_fuzz.cc [hunk 7]
```diff
@@ -413,7 +413,7 @@ uint32_t bufferAction(Context& ctxt, char insert_value, uint32_t max_alloc, Buff
 
 void executeActions(const test::common::buffer::BufferFuzzTestCase& input, BufferList& buffers,
                     BufferList& linear_buffers, Context& ctxt) {
-  // Soft bound on the available memory for allocation to avoid OOMs and
+  // Soft bound on the available memory for allocation to avoid `OOMs` and
```
Comment (ggreenway) [inline test/common/buffer/buffer_fuzz.cc:416]: Change to "OOM events" and it won't need to be quoted

File: test/common/config/delta_subscription_state_test.cc [hunk 8]
```diff
@@ -189,7 +189,7 @@ TEST_F(DeltaSubscriptionStateTest, CumulativeUpdates) {
 }
 
 // Verifies that a sequence of good and bad responses from the server all get the appropriate
-// ACKs/NACKs from Envoy.
+// ACKs/`NACKs` from Envoy.
```
Comment (ggreenway) [inline test/common/config/delta_subscription_state_test.cc:192]: I think if ACK is in the dictionary, NACK should also be. It's common enough to be easily understood.

File: test/common/http/conn_manager_utility_test.cc [hunk 9]
```diff
@@ -739,7 +739,7 @@ TEST_F(ConnectionManagerUtilityTest, ExternalAddressInternalRequestUseRemote) {
   EXPECT_EQ("10", headers.get_("x-envoy-expected-rq-timeout-ms"));
 }
 
-// Make sure we don't remove connection headers for WS requests.
+// Make sure we don't remove connection headers for `WS` requests.
```
Comment (ggreenway) [inline test/common/http/conn_manager_utility_test.cc:742]: replace with websocket

File: test/common/http/conn_manager_utility_test.cc [hunk 10]
```diff
@@ -749,7 +749,7 @@ TEST_F(ConnectionManagerUtilityTest, DoNotRemoveConnectionUpgradeForWebSocketReq
   EXPECT_EQ("websocket", headers.get_("upgrade"));
 }
 
-// Make sure we do remove connection headers for non-WS requests.
+// Make sure we do remove connection headers for `non-WS` requests.
```
Comment (ggreenway) [inline test/common/http/conn_manager_utility_test.cc:752]: non-websocket

**Review by ggreenway — CHANGES_REQUESTED** [review]
There's also a clang-tidy failure. clang-tidy runs on a file after it's touched, so even though it's unrelated to this change, you'll have to fix it.

/wait

File: tools/spelling/spelling_dictionary.txt [hunk 11]
```diff
@@ -151,25 +146,19 @@ IPV
 IPs
 IPv
 ITOA
-Injectable
```
[the remark this answers is not in the document]
Comment (mk46) [reply tools/spelling/spelling_dictionary.txt:154]: Done.

File: test/common/router/config_impl_test.cc [hunk 12]
```diff
@@ -2621,7 +2621,7 @@ TEST_F(RouterMatcherFilterStateHashPolicyTest, NullValue) {
   EXPECT_FALSE(route->routeEntry()->hashPolicy()->generateHash(nullptr, headers_, add_cookie_nop_,
                                                                filter_state_));
 }
-// Nonhashable.
+// `Nonhashable`.
```
[the remark this answers is not in the document]
Comment (mk46) [reply test/common/router/config_impl_test.cc]: I think it should be `Nonhashable key`.

File: tools/spelling/spelling_dictionary.txt [hunk 13]
```diff
@@ -201,53 +188,34 @@ NOLINT
...
 OU
 OVFL
 PAYLOADLEN
 PB
-PCC
-PEERCRED
 PEM
 PERF
 PGV
-PID
```
[the remark this answers is not in the document]
Comment (mk46) [reply tools/spelling/spelling_dictionary.txt:229]: Done.

File: source/extensions/filters/http/cache/http_cache.h [hunk 14]
```diff
@@ -46,7 +46,7 @@ class RawByteRange {
 public:
   // - If first==UINT64_MAX, construct a RawByteRange requesting the final last body bytes.
   // - Otherwise, construct a RawByteRange requesting the [first,last] body bytes.
-  // Prereq: first == UINT64_MAX || first <= last
+  // `Prereq`: first == UINT64_MAX || first <= last
```
[the remark this answers is not in the document]
Comment (mk46) [reply source/extensions/filters/http/cache/http_cache.h:49]: Changed

File: tools/spelling/spelling_dictionary.txt (hunk 1 above)
Comment (mk46) [reply tools/spelling/spelling_dictionary.txt:241]: Done.

File: source/common/signal/signal_action.h (hunk 2 above)
Comment (mk46) [reply source/common/signal/signal_action.h]: Done.

File: tools/spelling/spelling_dictionary.txt (hunk 3 above)
Comment (mk46) [reply tools/spelling/spelling_dictionary.txt:265]: Reverted RW

File: source/extensions/quic_listeners/quiche/platform/quic_logging_impl.cc (hunk 4 above)
Comment (mk46) [reply source/extensions/quic_listeners/quiche/platform/quic_logging_impl.cc]: Done.

File: source/extensions/transport_sockets/tls/utility.cc (hunk 5 above)
Comment (mk46) [reply source/extensions/transport_sockets/tls/utility.cc:40]: Spell check fails.

File: source/server/connection_handler_impl.cc (hunk 6 above)
Comment (mk46) [reply source/server/connection_handler_impl.cc]: Done

File: test/common/buffer/buffer_fuzz.cc (hunk 7 above)
Comment (mk46) [reply test/common/buffer/buffer_fuzz.cc]: Done

File: test/common/config/delta_subscription_state_test.cc (hunk 8 above)
Comment (mk46) [reply test/common/config/delta_subscription_state_test.cc]: Reverted Nacks

File: test/common/http/conn_manager_utility_test.cc (hunk 9 above)
Comment (mk46) [reply test/common/http/conn_manager_utility_test.cc]: Done.

File: test/common/http/conn_manager_utility_test.cc (hunk 10 above)
Comment (mk46) [reply test/common/http/conn_manager_utility_test.cc]: Done.

**Review by lizan — CHANGES_REQUESTED** [review]
What's the purpose of this PR? Many words removed from dictionary is a valid use case and I don't want to back quote them every time.

e.g.:
- GSS
- GURL
- JSONs
- OAuth
- RECV*
- SIG*
- STDSTRING
- STS
- Symbolizer
- TE
- TXT
- WAVM

**Comment by ggreenway** [comment]
@lizan #12807 was filed because the dictionary file is filling up with one-use words, and a lot of them are things like a variable name. I think it's relatively clear that variable names shouldn't be in the dictionary.

It gets less clear with other technical terms. Aside from the ones I requested reverted, I thought what was being taken out in this one was a reasonable set, but a lot of it is just a matter of opinion. Maybe we should take a step back and figure out what rules we want to apply to what should be in the dictionary.

**Comment by mk46** [comment]
> There's also a clang-tidy failure. clang-tidy runs on a file after it's touched, so even though it's unrelated to this change, you'll have to fix it.
> 
> /wait

`source/extensions/quic_listeners/quiche/platform/quic_pcc_sender_impl.h` is implementation of [quic/platform/api/quic_pcc_sender.h#14](https://quiche.googlesource.com/quiche/+/master/quic/platform/api/quic_pcc_sender.h#14). Clang-tidy error occurs due method naming convention differs. Can we escape clang-tidy check for this?

**Comment by mk46** [comment]
@ggreenway PTAL.

**Review by ggreenway — COMMENTED** [review]
@mk46 As noted by @lizan, I think some of these changes end up being more work than they're worth. I don't think it's always clear which terms should be in the dictionary or not, but it's generally clearer to the person writing the code. It's less clear in a PR like this that isn't specific to any area of code.

On one hand, I don't really like that we have 1200 terms in our dictionary file. On the other hand, I don't know if it's worth the time to try to clean them out or not. If we are going to clean them out, I think we should make the spell checker not add them automatically or we'll get right back to where we're at now very quickly. 

If we do pursue this, I don't think it makes sense to remove the plural of a term but leave the singular, eg `JSON` is left but `JSONs` is removed.

**Pull request closed without merging**